### PR TITLE
Adding more p2p scenarios

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #.
https://github.com/couchbaselabs/mobile-testkit/issues/1063

- [x] Ran `flake8`

#### Changes proposed in this pull request:
https://github.com/couchbaselabs/mobile-testkit/issues/1063

### CODE REVIEW

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 1]
```diff
@@ -265,10 +335,182 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+    log_info("ls_url_one: {}".format(ls_url_one))
+    log_info("ls_url_two: {}".format(ls_url_two))
+
+    client = MobileRestClient()
+
+    log_info("Creating databases")
+    ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
+    ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
+
+    ls_db2_docs = client.add_docs(url=ls_url_two, db=ls_db2, number=num_docs_per_db, id_prefix="test_ls_db2")
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Use bulk_ops here.

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 2]
```diff
@@ -265,10 +335,182 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+    log_info("ls_url_one: {}".format(ls_url_one))
+    log_info("ls_url_two: {}".format(ls_url_two))
+
+    client = MobileRestClient()
+
+    log_info("Creating databases")
+    ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
+    ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
+
+    ls_db1_docs = client.add_docs(url=ls_url_one, db=ls_db1, number=num_docs_per_db, id_prefix="test_ls_db1")
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Use bulk_ops here.

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 3]
```diff
@@ -265,10 +335,182 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+    ls_db2_docs = client.add_docs(url=ls_url_two, db=ls_db2, number=num_docs_per_db, id_prefix="test_ls_db2")
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Use bulk_ops here.

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 4]
```diff
@@ -265,10 +335,182 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+    log_info("ls_url_one: {}".format(ls_url_one))
+    log_info("ls_url_two: {}".format(ls_url_two))
+
+    client = MobileRestClient()
+
+    log_info("Creating databases")
+    ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
+    ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
+
+    ls_db1_docs = client.add_docs(url=ls_url_one, db=ls_db1, number=num_docs_per_db, id_prefix="test_ls_db1")
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Use bulk_ops here.

**Comment by sethrosetter** [comment]
@raghusarangapani The Readmes should still be present

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 3 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Added the code but have commented it for now as addition through _bulk_docs needs more research

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 2 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Added the code but have commented it for now as addition through _bulk_docs needs more research

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 4 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Added the code but have commented it for now as addition through _bulk_docs needs more research

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 1 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Added the code but have commented it for now as addition through _bulk_docs needs more research

**Comment by raghusarangapani** [comment]
Change to use bulk_docs has been pushed

**Comment by raghusarangapani** [comment]
Ran the entire testsuites/listener/shared/client_client suite between net-mono clients - 100% pass

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 5]
```diff
@@ -210,8 +212,11 @@ def test_peer_2_peer_sanity_pull(setup_p2p_test):
     ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
     ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
 
-    # Setup continuous pull replication from LiteServ 2 ls_db2 to LiteServ 1 ls_db1
+    if seeded_db:
+        ls_db2_docs_seed = client.add_docs(url=ls_url_two, db=ls_db2, number=num_docs_per_db, id_prefix="test_ls_db2_seed")
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Please use bulk_docs here as well

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 6]
```diff
@@ -246,7 +258,60 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+    log_info("ls_url_two: {}".format(ls_url_two))
+
+    client = MobileRestClient()
+
+    log_info("Creating databases")
+    ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
+    ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
+
+    if seeded_db:
+        ls_db1_docs_seed = client.add_docs(url=ls_url_one, db=ls_db1, number=num_docs_per_db, id_prefix="test_ls_db1_seed")
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Please use bulk_docs here

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 7]
```diff
@@ -224,16 +229,23 @@ def test_peer_2_peer_sanity_pull(setup_p2p_test):
     ls_db2_docs = client.add_docs(url=ls_url_two, db=ls_db2, number=num_docs_per_db, id_prefix="test_ls_db2")
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Should we wait for replication status "Idle" here before adding more docs?

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 8]
```diff
@@ -257,6 +322,12 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
     ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
     ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
 
+    if seeded_db:
+        ls_db1_docs_seed = client.add_docs(url=ls_url_one, db=ls_db1, number=num_docs_per_db, id_prefix="test_ls_db1_seed")
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Please use bulk ops

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 9]
```diff
@@ -257,6 +322,12 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+        assert len(ls_db1_docs_seed) == num_docs_per_db
+        ls_db2_docs_seed = client.add_docs(url=ls_url_two, db=ls_db2, number=num_docs_per_db, id_prefix="test_ls_db2_seed")
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: please use bulk ops

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 10]
```diff
@@ -265,10 +336,255 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+    log_info("ls_url_two: {}".format(ls_url_two))
+
+    client = MobileRestClient()
+
+    log_info("Creating databases")
+    ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
+    ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
+
+    bulk_docs = create_docs("test_ls_db2", num_docs_per_db)
+    ls_db2_docs = client.add_bulk_docs(url=ls_url_two, db=ls_db2, docs=bulk_docs)
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Assert number of docs

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 11]
```diff
@@ -265,10 +336,255 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+    log_info("ls_url_two: {}".format(ls_url_two))
+
+    client = MobileRestClient()
+
+    log_info("Creating databases")
+    ls_db1 = client.create_database(url=ls_url_one, name="ls_db1")
+    ls_db2 = client.create_database(url=ls_url_two, name="ls_db2")
+
+    bulk_docs = create_docs("test_ls_db1", num_docs_per_db)
+    ls_db1_docs = client.add_bulk_docs(ls_url_one, ls_db1, bulk_docs)
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: assert number of docs

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 12]
```diff
@@ -265,10 +336,255 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: assert number of docs

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 13]
```diff
@@ -265,10 +336,255 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+
+    bulk_docs = create_docs("test_ls_db2", num_docs_per_db)
+    ls_db2_docs = client.add_bulk_docs(ls_url_two, ls_db2, bulk_docs)
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: assert number of docs

Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: It was there down below, moved it up

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 12 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: It was there down below, moved it up

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 11 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: It was there down below, moved it up

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 10 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: It was there down below, moved it up

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 9 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Done

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 8 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Done

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 6 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Done

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 5 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Done

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py (hunk 7 above)
Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: I wasn't sure. Removed it.

Comment (raghusarangapani) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Done

File: testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py [hunk 14]
```diff
@@ -265,10 +341,252 @@ def test_peer_2_peer_sanity_push(setup_p2p_test):
...
+    # Setup one shot push replication from LiteServ 1 ls_db1 to LiteServ 2 ls_db2
+    log_info("Setting up a one-shot push replication from ls_db1 to ls_db2")
+    push_repl = client.start_replication(
+        url=ls_url_one,
+        continuous=False,
+        from_db=ls_db1,
+        to_url=ls_url_two, to_db=ls_db2,
+    )
+    log_info("Replication ID: {}".format(push_repl))
+    time.sleep(5)
```
Comment (sethrosetter) [inline testsuites/listener/shared/client_client/test_peer_2_peer_sanity.py]: Is this necessary?

**Comment by raghusarangapani** [comment]
Re-ran the entire testsuites/listener/shared/client_client suite between net-mono clients - 100% pass

**Pull request merged**